infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesSnapshotReader.java

```diff
@@ -197,9 +196,17 @@ public SnapshotReadMessage read(UUID syncRequestId) {
         log.info("Start Snapshot Sync replication for stream name={}, id={}", streamNameOfTag,
                 CorfuRuntime.getStreamID(streamNameOfTag));
 
-        // Wait for data to be available on the Snapshot Sync Queue if it is empty
+        // If data is not present in the snapshot sync queue:
+        // (1) check if data for any session was received in the last time window. If not, request for snapshot sync
+        // data from the client again.
+        // (2) If within the timeout window, enqueue a snapshot_continue event. This would mean that the client is
+        // providing data for other sessions, and is yet to start generating for the current session.
         if (currentStreamInfo == null || !currentStreamHasNext()) {
-            waitForData();
+            if (!dataFound() && receiveWindowTimedOut()) {
```

Comment:
can we avoid 2 calls to dataFound()?  Something like - if (!dataFound()) { if (receiveWindowTimedOut())...}}